test/test_optim.py

```diff
@@ -477,6 +477,88 @@ def test_multi_tensor_optimizers(self):
+            # ((optim.NAdam), dict(weight_decay=1., momentum_decay=6e-3)),
+            # ((optim.NAdam), dict(weight_decay=0., momentum_decay=4e-3)),
+            # ((optim.NAdam), dict(weight_decay=0.01, momentum_decay=4e-3)),
+            # ((optim.SGD), dict(lr=0.2, momentum=1, dampening=0, weight_decay=1, nesterov=True)),
+            # ((optim.SGD), dict(lr=0.2, momentum=1, dampening=0.5, weight_decay=1, nesterov=False)),
+            # ((optim.RAdam), dict(weight_decay=0)),
+            # ((optim.RAdam), dict(weight_decay=1)),
+            # ((optim.RMSprop), dict(weight_decay=1, momentum=1, centered=True)),
+            # ((optim.RMSprop), dict(weight_decay=1, momentum=0, centered=True)),
+            # ((optim.RMSprop), dict(weight_decay=1, momentum=1, centered=False)),
+            # ((optim.RMSprop), dict(weight_decay=0, momentum=1, centered=False)),
+            # ((optim.Rprop), dict(lr=1e-2, etas=(0.5, 1.2), step_sizes=(1e-6, 50))),
+            # ((optim.ASGD), dict(weight_decay=0)),
+            # ((optim.ASGD), dict(weight_decay=1)),
+            # ((optim.Adamax), dict(weight_decay=0)),
+            # ((optim.Adamax), dict(weight_decay=1)),
+            ((optim.Adadelta), dict(weight_decay=0)),
+            ((optim.Adadelta), dict(weight_decay=1)),
+            # ((optim.Adagrad), dict(weight_decay=0)),
+            # ((optim.Adagrad), dict(weight_decay=1)),
+        ]
+
+        kIterations = 11
+        device = 'cuda'
+
+        for opt, params in optimizer_pairs_with_flags:
+            res = []
+            for foreach in (True, False):
+                weight = torch.tensor([[-0.2109, -0.4976], [-0.1413, -0.3420], [-0.2524, 0.6976]],
+                                      dtype=torch.float64, device=device, requires_grad=True)
+                bias = torch.tensor([-0.1085, -0.2979, 0.6892], dtype=torch.float64, device=device, requires_grad=True)
+                weight2 = torch.tensor([[-0.0508, -0.3941, -0.2843]],
+                                       dtype=torch.float64, device=device, requires_grad=True)
+                bias2 = torch.tensor([-0.0711], dtype=torch.float64, device=device, requires_grad=True)
+                input = torch.tensor([0.1, 0.2, 0.3, 0.4, 0.5, 0.6], dtype=torch.float64, device=device).reshape(3, 2)
+
+                model = torch.nn.Sequential(torch.nn.Linear(2, 3),
+                                            torch.nn.Sigmoid(),
+                                            torch.nn.Linear(3, 1),
+                                            torch.nn.Sigmoid())
+                model.to(torch.float64).to(device)
+
+                pretrained_dict = model.state_dict()
+                pretrained_dict['0.weight'] = weight
+                pretrained_dict['0.bias'] = bias
+                pretrained_dict['2.weight'] = weight2
+                pretrained_dict['2.bias'] = bias2
+                model.load_state_dict(pretrained_dict)
+
+                params['foreach'] = foreach
```

Comment:
Aren't you modifying the dictionary contained in optimizer_pairs_with_flags here?
Maybe it doesn't matter as you never re-use it but it looks a bit dangerous no?